Implement Front End of PostIt

phemmz opened the pull request:
#### What does this PR do?
- Improve on front-end template built in challenge 0 using Material Design Framework.
- Setup and implement front-end application using ReactJS-Redux.
- Configure webpack for transpiling and converting SCSS to CSS
- Consume api built in previous challenge
- Tests for actions, reducers and components using Enzyme and Jest
#### How should this be manually tested?
- This can be tested by accessing the app on https://phemmz-post-it.herokuapp.com

phemmz committed:
fix travis build error

phemmz committed:
fix dialect error on travis

phemmz committed:
integrate test coverage reporting

phemmz committed:
add coverall badge

phemmz committed:
configure validation and authentication

phemmz committed:
add session to getgroup function

phemmz committed:
add more tests for routes

phemmz committed:
update readme to include payloads

phemmz committed:
update readme

phemmz committed:
update readme file for API documettation

phemmz committed:
fix validator errors

phemmz committed:
fix validations not re running

phemmz committed:
configure input validations for all routes

phemmz committed:
write more api tests

phemmz committed:
refactor template design

phemmz committed:
fix associations not working in models

phemmz committed:
fix error missing column for usergroup model

phemmz committed:
integrate function to check if user exist before addiing a user to a group

phemmz committed:
fix post message route not working

phemmz committed:
remove validation for checking if a user belongs to a group before adding

phemmz committed:
remove config.json

phemmz committed:
refactor test codes

phemmz committed:
update readme file

phemmz committed:
implement group member validation for users

phemmz committed:
write more tests to pass

phemmz committed:
improve template design with materialize

phemmz committed:
Render main react component

phemmz committed:
Hot-reloading setup

phemmz committed:
React router and basic navigation between pages

phemmz committed:
Sends data to server via redux thunk action

phemmz committed:
Add server side validations

phemmz committed:
Adds client side validation

phemmz committed:
Adds flash messages reducer

phemmz committed:
Renders flash messages and allows to delete them

phemmz committed:
Validates username and email uniqueness on server

phemmz committed:
Validate username and email uniqueness on client and server side

phemmz committed:
Create login form

phemmz committed:
Verify user credentials

phemmz committed:
Use jwt on client and set current user in redux store

phemmz committed:
Login and logout on client

phemmz committed:
Authenticate express middleware

phemmz committed:
requireAuth higher order component

phemmz committed:
Create a component for posting messages to a group

phemmz committed:
Create presentational components for goroup and message

phemmz committed:
View messages in a group

phemmz committed:
Configure webpack hot middleware and add jsdoc comments

phemmz committed:
Add jsdoc comments and refactor code to use javascript class

phemmz committed:
Add jsdoc comments

phemmz committed:
Add jsdoc comments and refactor code to use javascript class

phemmz committed:
Add jsdoc comments and refactor code to use javascript class

phemmz committed:
Add action constants

phemmz committed:
Create textfield group component

phemmz committed:
Delete footer.js component

phemmz committed:
Connect component to redux and implement other group functions

phemmz committed:
Delete Header component

phemmz committed:
Connect component to redux and implement other message functionalities

phemmz committed:
Redesign layout to be more responsive

phemmz committed:
Change materialize grid column class

phemmz committed:
Set layout to include navigation bar and footer

phemmz committed:
Add jsdoc comments and connect component to redux

phemmz committed:
Add jsdoc comment and connect component to redux

phemmz committed:
Redesign welcome container on landing page

phemmz committed:
Implement flash message

phemmz committed:
Implement flash message

phemmz committed:
Connect component to redux, add jsdoc comment and set up router

phemmz committed:
Add materialize modal and redesign create group component

phemmz committed:
Implement flash messages

phemmz committed:
Create signup component and implement the functionality

phemmz committed:
Redesign presentational component for creating a message

phemmz committed:
Add jsdoc comment and implement onSelect feature

phemmz committed:
Add message creator to message displayed

phemmz committed:
Create reducer for authentication action

phemmz committed:
Create reducer for flash messages action

phemmz committed:
Create reducer for group  actions

phemmz committed:
Set authorization token in the header

phemmz committed:
Install new dependencies packages

phemmz committed:
Remove file

phemmz committed:
Add jsdoc comments

phemmz committed:
Add jsdoc comments

phemmz committed:
Remove index file

phemmz committed:
Add jsdoc comments

phemmz committed:
Add jsdoc comments

phemmz committed:
Add jsdoc comments

phemmz committed:
Add jsdoc comments

phemmz committed:
Add jsdoc comments

phemmz committed:
Add jsdoc comments

phemmz committed:
Add jsdoc comments

phemmz committed:
Add jsdoc comments

phemmz committed:
Add column for message creator

phemmz committed:
Add column for message creator

phemmz committed:
Add jsdoc comments and change to export default

phemmz committed:
Add jsdoc comments and change to export default

phemmz committed:
Add jsdoc comments and change to export default

phemmz committed:
Add jsdoc comments and change to export default

phemmz committed:
Add jsdoc comments

phemmz committed:
Remove index file

phemmz committed:
Add jsdoc comments

phemmz committed:
Add jsdoc comments

phemmz committed:
Add links for materialize icons and font

phemmz committed:
Add new styles

phemmz committed:
Remove old logo pic

phemmz committed:
Add favicon

phemmz committed:
Configure webpack for hosting

phemmz committed:
Add babel-plugin-react-transform

phemmz committed:
Change background color for messages layout

phemmz committed:
Add getUsers action and jsdoc comment for description

phemmz committed:
Add action constant for getting users

phemmz committed:
Add jsdoc description comment

phemmz committed:
Add html validation for input fields

phemmz committed:
Add materialize toast for flash messages

phemmz committed:
Populate select button options with users from database

phemmz committed:
Add materialize toast for succesful login

phemmz committed:
Change input field type for email

phemmz committed:
Add reducer for getting users

phemmz committed:
Remove .DS_Store file

phemmz committed:
Change design for group details side nav

phemmz committed:
Add email to token payload

phemmz committed:
Add design for side nav

phemmz committed:
[Feature jwt-on-signup] Generate token on successful signup and save in localStorage

phemmz committed:
[feature issue-jwt-on-signup] Remove commented files for side nav

phemmz committed:
[feature issue-jwt-on-signup] Add jsdoc comments

phemmz committed:
[Feature issue-jwt-on-signup] Replace flashmessage with materialize toast and redirect to dashboard on successful signup

phemmz committed:
[Feature issue-jwt-on-signup] Generate token on server side on signup

phemmz committed:
Render sidenav and add user modal presentational components

phemmz committed:
Change message display to email format

phemmz committed:
Add sidenav and usermodal to index export

phemmz committed:
Create Add user modal presentational component

phemmz committed:
Create SideNav presentational component

phemmz committed:
Add height for container displaying messages

phemmz committed:
Fix heroku error

phemmz committed:
Specify node version in package.json

phemmz committed:
Change jwt-decode version

phemmz committed:
Add jwt to tests

phemmz committed:
authenticate route for getting all users

phemmz committed:
Remove session authentication

phemmz committed:
Remove session authentication

phemmz committed:
Use jwt decoded current user details

phemmz committed:
Change error message

phemmz committed:
Change error message

phemmz committed:
Remove session authentication

phemmz committed:
Use jwt decoded user details

phemmz committed:
Use jwt decoded user details

phemmz committed:
Add jsdoc comments

phemmz committed:
Install redux-mock-store package

phemmz committed:
Add reducer test for reducers

phemmz committed:
Add action test for SET_CURRENT_USER

phemmz committed:
Add jsdoc comment

phemmz committed:
Remove addFlashMessage action and replace with materialize toast

phemmz committed:
Remove addFlashMessage

phemmz committed:
Add reducer for sendMessage action

phemmz committed:
Remove flashMessages reducer

phemmz committed:
Remove CreateGroup from index file

phemmz committed:
Add jsdoc comment and add mouse pointer to group anchor tags

phemmz committed:
Add jsdoc comments

phemmz committed:
Remove CreateGroup component

phemmz committed:
Add jsdoc comments

phemmz committed:
Change error message

phemmz committed:
Add jsdoc comments

phemmz committed:
Remove flasmessage

phemmz committed:
Remove flashmessages

phemmz committed:
Add jsdoc comments

phemmz committed:
Add jsdoc comments

phemmz committed:
Add sass/scss syntax

phemmz committed:
Add return to api call

phemmz committed:
Add sendMessage action constant

phemmz committed:
Change styling for displaying error messages

phemmz committed:
Add jsdoc comments and implement postMessage action

phemmz committed:
Create a route path for creating group page

phemmz committed:
Add CreateGroup container

phemmz committed:
Test for actions and reducers

phemmz committed:
client bundled files

phemmz committed:
Resolve merge conflict

phemmz committed:
bundled files

phemmz committed:
install modules for testing react components

phemmz committed:
Make dashboard more responsive

phemmz committed:
Make dashboard more responsive

phemmz committed:
change webpack to es6

phemmz committed:
Fix webpack error

phemmz committed:
feat: implement reset password
- ensure users get verified before changing password
- update password field for the user
[Finishes #150368936]

phemmz committed:
feat: implement reset password
- change settings for running migration to database
[Finishes ##150368936]

phemmz committed:
feat: implement reset password for users
- change eslint esversion in config file
[Finishes #150368936]

phemmz committed:
feat: implement reset password for users
- install peer dependencies for eslint
[Finishes #150368936]

phemmz committed:
Merge pull request #5 from phemmz/feat/150368936/reset-password-for-users

#150368936 Implement reset password for users

phemmz committed:
feat: implement google authentication
- render react google login component to connect to google
- create a google app account
- generate a token on response from google and save the user details to the databse
[Finishes #150370554]

phemmz committed:
feat: implement google authentication
- fix eslint errors
[Finishes #150370554]

phemmz committed:
Merge pull request #7 from phemmz/feat/150370554/google-authentication

#150370554 Implement google authentication

phemmz committed:
bug: implement all feedback
- configure eslint to work for react files
- fix all eslint errors
- use jsx as file extension
- change icons
[Finishes #150474933]

phemmz committed:
fix: implement feedback
- add window to eslint config as global variable
- clear errors in state after completing an action
[Finishes #150474933]

phemmz committed:
fix: implement feedback
- add configuration for houndci
[Finishes #15047933]

phemmz committed:
fix: implement feedback
- fix houndci resolve path error
[Finishes #150474933]

phemmz committed:
fix: implement feedback
- fix houndci import error
[Finishes #150474933]

phemmz committed:
Merge pull request #11 from phemmz/bug/150474933/implement-feedback

#150474933 Implement feedback

phemmz committed:
feat: implement in app notification
- ignore dist folder and nodemodules files in eslint
- emit a new msg event from the server side and listen for the event on client
- use event data on the client side to show notification for users
[Finishes #150375415]

phemmz committed:
feat: implement in app notification
- fix houndci errors
[Finishes #150375415]

phemmz committed:
Merge pull request #13 from phemmz/feat/150375415/notification-for-new-messages

#150375415 Implement notification for new messages

phemmz committed:
feat: priority-level
- implement email and sms notifications for different message priority level
[Finishes #150375883]

phemmz committed:
Merge pull request #15 from phemmz/feat/150375883/priority-level

#150375883 Implement priority-level

phemmz committed:
feat: implement read list
- create a table that saves users who have seen the message
- fire the action that makes the call to the api when user view the message
- populate tooltip with users who have seen the message
[Finishes #150376139]

phemmz committed:
feat: implement read list
- fix houndci errors
[Finishes #150376139]

phemmz committed:
Merge pull request #18 from phemmz/feat/150376139/read-list-for-users

#150376139 Implement read list for users

phemmz committed:
feat: implement search for users
- create a new page where users can search for other users
- query database to find any user that has the value the user enters
- creat an action that makes the call to the database
- loop through the response from the call and display it to the user
[Finishes #150376484]

phemmz committed:
Merge pull request #19 from phemmz/feat/150376484/search-for-users

#150376484 Implement feature for users to search for other users

phemmz committed:
bug: implement second feedback
- create a .env.sample file
- protect login and signup page when user is logged in
- version all routes
- use event instead of e
[Finishes #150806984]

phemmz committed:
bug: fix all feedback from review
- fix travis
[Finishes #150806984]

phemmz committed:
Merge pull request #20 from phemmz/bug/150806984/implement-second-review-feedback

#150806984 implement second review feedback

phemmz committed:
feat: implement pagination
- install react paginate and configure it
- style searched results and display it using pagination
[Finishes #150376484]

phemmz committed:
feat: implement pagination
- fix houndci errors
[Finishes #150376484]

phemmz committed:
feat: implement pagination
- fix scss houndci errors
[Finishes #150376484]

phemmz committed:
Merge pull request #23 from phemmz/feat/150376484/implement-pagination

#150376484 Implement pagination

phemmz committed:
chore: tests for components
- write tests for components
- write more tests for server side to improve coverage
- clean up code
[Finishes #150899409]

phemmz committed:
chore: write tests for components
- fix houndci comments on PR
[Finishes #150899409]

phemmz committed:
Merge pull request #25 from phemmz/chore/150899409/tests-for-components

#150899409 Write tests for components

blessingoraz commented on client/src/components/containers/Groups.jsx:
```diff
@@ -0,0 +1,166 @@
+    const listItems = this.props.groupList.map((group, index) => {
+      const selected = (group.id === Number(this.props.selectedGroup));
+      return (
+        <li className="each-group" key={index}>
```
Eslint is throwing this warning because it sees the use of index as an anti-pattern and this might break your code application. 

A solution should be giving a unique property to your items which should be `id`.

phemmz replied:
I tried using id but i was getting this error. But, i will look at it again and try to fix this. Thanks

blessingoraz commented on client/src/components/containers/CreateGroup.jsx:
```diff
@@ -0,0 +1,109 @@
+                  <div className="input-field col s12">
+                    <i className="material-icons prefix">group</i>
+                    <input
+                      ref="group"
```
You can avoid this warning string literals by using  `ref={(el) => this.groupRef = el}`

phemmz replied:
Noted. Will do that.

phemmz committed:
fix: fix tooltip
- fix tooltip error
[Finishes #151016504]

phemmz committed:
fix: fix tooltip
- write new test for tooltip fix
[Finishes #151016504]

phemmz committed:
Merge pull request #27 from phemmz/bug/151016504/fix-tooltip

#151016504 Fix tooltip

phemmz committed:
feat: implement group members
- create route for group members page
- create an action that makes the call to the api
- render a component that displays the members
- write tests
[Finishes #150992586]

phemmz committed:
Merge pull request #28 from phemmz/feat/150992586/members-in-group

#150992586 Implement group members

phemmz committed:
fix: fix props error
- fix props error when group is created
[Finishes #150992586]

phemmz committed:
Merge pull request #29 from phemmz/bug/150992586/fix-props-error

fix: fix props error

phemmz committed:
bug: implement feedback
- write more test for components
- put action for getting users and searching users into useraction files
[Finishes #150806984]

phemmz committed:
Merge pull request #30 from phemmz/bug/150806984/implement-second-review-feedback

#150806984 Implement feedback

phemmz committed:
chore: setup production webpack
- setup webpack for production build of react
[Finishes #151160877]

phemmz committed:
Merge pull request #31 from phemmz/chore/151160877/webpack-production

#151160877 Setup production webpack

phemmz committed:
chore: tests for components
- refactor client side tests
[Finishes #150899409]

phemmz committed:
Merge pull request #32 from phemmz/chore/150899409/tests-for-components

#150899409 Tests for components

phemmz committed:
chore: tests-for-components
- add coverage files to gitignore
[Finishes #150899409]

phemmz committed:
Merge pull request #33 from phemmz/chore/150899409/tests-for-components

#150899409 tests-for-components

phemmz committed:
bug: implement feedback
- change implementation of add users to group feature
[Finishes #150474933]

phemmz committed:
fix: implement feedback
- review add user feature to make use of react chips
- Change flash message when no message is entered
- catch error that occurs when string is supplied as groupId in api
- move all test mock data to a separate file
- fix multiple notification messages when a message is posted
[Finishes #150474933]

phemmz committed:
fix: implement feedback
- fix houndci errors
[Finishes #150474933]

phemmz committed:
Merge pull request #34 from phemmz/bug/150474933/implement-feedback

#150474933 Implement feedback

phemmz committed:
fix: reset password issue
- clear state on failed reset password attempt
[Finishes #150474933]

phemmz committed:
fix: fix reset password issue
- fix coverage error
[Finishes #150474933]

phemmz committed:
Merge pull request #35 from phemmz/bug/150474933/implement-feedback

#150474933 reset password issue

phemmz committed:
fix: implement feedback
- write more server tests
[Finishes #150474933]

phemmz committed:
Merge pull request #36 from phemmz/bug/150474933/implement-feedback

#150474933 implement feedback

phemmz committed:
Update README.md

phemmz committed:
fix: implement feedback
- pass groupmessages through redux store
[Finishes #150474933]

phemmz committed:
Merge pull request #37 from phemmz/bug/150474933/implement-feedback

#150474933 Implement feedback

phemmz committed:
fix: implement feedback
- install sinon
- write more tests for components
- refactor code
[Finishes #150474933]

phemmz committed:
Merge pull request #38 from phemmz/bug/150474933/implement-feedback

#150474933 implement feedback

phemmz committed:
Rename requireAuth.jsx to RequireAuth.jsx

phemmz committed:
Rename app.jsx to App.jsx

phemmz committed:
fix: implement feedback
- move test mock data to a separate file for reducer tests
[Finishes #150474933]

phemmz committed:
fix: implement feedback
- refactor middlewares for server side
- change data to an appropriate name
- move all test mock data to another file
[Finishes #150474933]

phemmz committed:
Merge pull request #39 from phemmz/bug/150474933/implement-feedback

#150474933 Implement feedback

phemmz committed:
fix: implement feedback
- delete validateCreateGroup in userValidations
- refactor client side test
[Finishes #150474933]

phemmz committed:
fix: implement feedback
- fix webpack error
[Finishes #150474933]

phemmz committed:
fix: implement feedback
- remove all server side transpiled files
[Finishes #150474933]

phemmz committed:
Merge pull request #40 from phemmz/bug/150474933/implement-feedback

#150474933 Implement feedback
